spring-kafka/src/main/java/org/springframework/kafka/listener/ContainerProperties.java

```diff
@@ -1116,56 +1116,108 @@ public void setRecordObservationsInBatch(boolean recordObservationsInBatch) {
+		appendEnabledProperty(sb, "idlePartitionEventInterval", this.idlePartitionEventInterval);
+
+		// Transaction management
+		appendOptionalProperty(sb, "transactionManager", this.transactionManager);
+		appendOptionalProperty(sb, "kafkaAwareTransactionManager", this.kafkaAwareTransactionManager);
+		appendOptionalProperty(sb, "transactionDefinition", this.transactionDefinition);
+
+		// Monitoring and scheduling
+		appendProperty(sb, "monitorInterval", this.monitorInterval);
+		appendOptionalProperty(sb, "scheduler", this.scheduler);
+		appendProperty(sb, "noPollThreshold", this.noPollThreshold);
+
+		// Container behavior flags
+		appendProperty(sb, "pauseImmediate", this.pauseImmediate);
+		appendProperty(sb, "stopImmediate", this.stopImmediate);
+		appendProperty(sb, "stopContainerWhenFenced", this.stopContainerWhenFenced);
+		appendProperty(sb, "asyncAcks", this.asyncAcks);
+
+		// Polling and partition configuration
+		appendProperty(sb, "pollTimeoutWhilePaused", this.pollTimeoutWhilePaused);
+		appendOptionalProperty(sb, "subBatchPerPartition", this.subBatchPerPartition);
+		appendProperty(sb, "assignmentCommitOption", this.assignmentCommitOption);
+		appendProperty(sb, "idleBetweenPolls", this.idleBetweenPolls);
+
+		// Header and recovery configuration
+		appendProperty(sb, "deliveryAttemptHeader", this.deliveryAttemptHeader);
+		appendProperty(sb, "batchRecoverAfterRollback", this.batchRecoverAfterRollback);
+
+		// Exactly-once semantics
+		appendProperty(sb, "eosMode", this.eosMode);
+
+		// Logging and error handling
+		appendProperty(sb, "logContainerConfig", this.logContainerConfig);
+		appendProperty(sb, "missingTopicsFatal", this.missingTopicsFatal);
+		appendProperty(sb, "restartAfterAuthExceptions", this.restartAfterAuthExceptions);
+
+		// Metrics and observation
+		appendProperty(sb, "micrometerEnabled", this.micrometerEnabled);
+		appendProperty(sb, "observationEnabled", this.observationEnabled);
+		appendProperty(sb, "recordObservationsInBatch", this.recordObservationsInBatch);
+		appendOptionalProperty(sb, "observationConvention", this.observationConvention);
+		appendOptionalProperty(sb, "observationRegistry", this.observationRegistry);
+
+		// Data multiplier
+		appendProperty(sb, "idleBeforeDataMultiplier", this.idleBeforeDataMultiplier);
+
+		sb.append("\n]");
+		return sb.toString();
+	}
+
+	/**
+	 * Append a property to the StringBuilder with consistent formatting.
+	 *
+	 * @param sb    the StringBuilder
+	 * @param name  the property name
+	 * @param value the property value
+	 */
+	private void appendProperty(StringBuilder sb, String name, Object value) {
+		sb.append("\n ").append(name).append("=").append(value);
+	}
+
+	/**
+	 * Append an optional property only if it's not null.
+	 *
+	 * @param sb    the StringBuilder
+	 * @param name  the property name
+	 * @param value the property value (nullable)
+	 */
+	private void appendOptionalProperty(StringBuilder sb, String name, @Nullable Object value) {
```

Comment:
I don't think we need separate methods for the logic.
Just one `appendProperty()` with `@Nullable Object value` and that `if` is enough.